lib/game_state.rb

```diff
@@ -8,17 +9,59 @@ class GameState
 	GUESS_LIMIT = 10
 	CODE_LENGTH = 4
 
-	attr_accessor :has_won, :past_guesses
-	attr_reader :secret_code 
 
-	def initialize
+	attr_accessor :has_won
+	attr_reader :secret_code, :past_guesses
+		
+	def initialize(sc = CodeGenerator, g = Guess)
 		@has_won = false
-		@past_guesses = PastGuesses.new
-		@secret_code = CodeGenerator.generate
+		@past_guesses = []
+		@secret_code = sc.generate
+		@g = Guess
 	end
 
 	def remaining_guesses
 		GUESS_LIMIT - past_guesses.size
 	end
 
+	def save_guess(guess)
+		past_guesses << guess
+	end
+
+	def save_state
+		current_state = { 
+			:has_won => @has_won, 
+			:secret_code => @secret_code,
+			:past_guesses => @past_guesses.map { |g| g.code }
+		}
```

Comment:
This file IO feels ripe for pulling out into another class. Maybe StateSaver?

Reply:
Because then the JSON dependency would be yanked out into a very thin file that only does a few things...

Reply:
Same with the state restoring later.